discord-scripts/invite-management.ts

```diff
@@ -133,10 +209,86 @@ export default async function sendInvite(discordClient: Client, robot: Robot) {
           }
         } catch (error) {
           robot.logger.error(
-            `An error occurred setting up the defense audit channel: ${error}`,
+            `An error occurred setting up the defense ${auditChannelType.toLowerCase()} audit channel: ${error}`,
           )
         }
       }
     })
+
+    // WIP, just testing out invite counting in order to verify which invite was used on join
+    discordClient.on("guildMemberAdd", async (member: GuildMember) => {
+      // for debugging
+      robot.logger.info(member)
+
+      const oldInvites =
+        (guildInvites.get(member.guild.id) as Collection<
+          string,
+          { uses: number }
+        >) || new Collection<string, { uses: number }>()
+      const fetchedInvites = await member.guild.invites.fetch()
+      const newInvites = new Collection<string, number>(
+        fetchedInvites.map((invite) => [invite.code, invite.uses ?? 0]),
+      )
+      guildInvites.set(member.guild.id, newInvites)
+
+      robot.logger.info(
```

Comment:
Will streamline this to reduce loggin for debugging and fix up `oldInvites` and `newInvites`